editor.planx.uk/src/@planx/components/PlanningConstraints/Modal.tsx

```diff
@@ -0,0 +1,247 @@
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import Divider from "@mui/material/Divider";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import visuallyHidden from "@mui/utils/visuallyHidden";
+import { Constraint, Metadata } from "@opensystemslab/planx-core/types";
+import omit from "lodash/omit";
+import React, { useState } from "react";
+import InputLabel from "ui/public/InputLabel";
+import ChecklistItem from "ui/shared/ChecklistItem";
+import ErrorWrapper from "ui/shared/ErrorWrapper";
+import Input from "ui/shared/Input";
+
+import { formatEntityName } from "./List";
+import { InaccurateConstraints } from "./Public";
+
+interface OverrideEntitiesModalProps {
+  showModal: boolean;
+  setShowModal: (value: React.SetStateAction<boolean>) => void;
+  fn: Constraint["fn"];
+  entities: Constraint["data"] | null;
+  metadata?: Metadata;
+  inaccurateConstraints: InaccurateConstraints;
+  setInaccurateConstraints: (
+    value: React.SetStateAction<InaccurateConstraints>,
+  ) => void;
+}
+
+const ERROR_MESSAGES = {
+  checklist: "Select at least one option",
+  input: "Enter a value",
+};
+
+export const OverrideEntitiesModal = ({
+  showModal,
+  setShowModal,
+  fn,
+  entities,
+  metadata,
+  inaccurateConstraints,
+  setInaccurateConstraints,
+}: OverrideEntitiesModalProps) => {
+  const [checkedOptions, setCheckedOptions] = useState<string[] | undefined>(
+    inaccurateConstraints?.[fn]?.["entities"],
+  );
+  const [showChecklistError, setShowChecklistError] = useState<boolean>(false);
+  const [textInput, setTextInput] = useState<string | undefined>(
+    inaccurateConstraints?.[fn]?.["reason"],
+  );
+  const [showInputError, setShowInputError] = useState<boolean>(false);
+
+  const title = `Which ${
+    metadata?.plural?.toLowerCase() || "entities"
+  } are inaccurate?`;
+
+  const closeModal = (_event: any, reason?: string) => {
+    if (reason && reason == "backdropClick") {
+      return;
+    }
+
+    // Clear any non-submitted inputs on cancel & sync parent state
+    setCheckedOptions(undefined);
+    setTextInput(undefined);
+    const newInaccurateConstraints = omit(inaccurateConstraints, fn);
+    setInaccurateConstraints(newInaccurateConstraints);
```

Comment:
This was a bit unexpected for a "cancel" interaction - I'd expect the initial state to persist, just changes I've made to be lost on cancel. This is bit more like a reset.

Reply:
Good catch - I was treating this like a "clear/reset". Now adjusted to reset to initial component state on "Cancel" :+1: